src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java

```diff
@@ -1282,9 +1291,22 @@ private boolean shouldReconnectToWriter(Boolean readOnly) {
    * @throws SQLException if an error occurs
    */
   @Override
-  protected synchronized void doClose() throws SQLException {
-    this.currentConnection.close();
-    releasePluginManager();
+  protected void doClose() throws SQLException {
+    synchronized (this.lockObject) {
+      try {
+        this.pluginManager.execute(this.getClass(), "close", () -> {
+          this.currentConnection.close();
+          return null;
+        });
+      }
+      catch(SQLException sqlEx) {
+        throw sqlEx;
+      }
+      catch(Exception ex) {
+        throw new SQLException(ex.getMessage(), ex);
+      }
+      releasePluginManager();
```

Comment:
`releasePluginManager();` as finally for `doClose()`, `doAbort`, & `doAbortInternal()`

Reply:
I'd like to avoid releasing plug manager if there's problems closing connection.